ionic-angular-template/src/services/auth.service.ts

```diff
@@ -35,18 +35,21 @@ export class AuthService {
 
   signIn() {
     return this.api.get(`${environment.apiUrl}/users/wallet`);
-}
+  }
+
+  async createUser(wallet: string, uid: string, email: string) {
+    return this.api.post(`${environment.apiUrl}/users/createUser`, {wallet, uid, email}).toPromise();
```

Comment:
You can put .toPromise() in the API, and not call it every time when you want to use post, get and etc.